lib/src/icc.cpp

```diff
@@ -432,49 +344,30 @@ std::shared_ptr<DataStruct> IccHelper::writeIccProfile(uhdr_color_transfer_t tf,
       // Should not fall here.
       return nullptr;
   }
-
-  // Compute primaries.
-  {
-    tags.emplace_back(kTAG_rXYZ,
-                      write_xyz_tag(toXYZD50.vals[0][0], toXYZD50.vals[1][0], toXYZD50.vals[2][0]));
-    tags.emplace_back(kTAG_gXYZ,
-                      write_xyz_tag(toXYZD50.vals[0][1], toXYZD50.vals[1][1], toXYZD50.vals[2][1]));
-    tags.emplace_back(kTAG_bXYZ,
-                      write_xyz_tag(toXYZD50.vals[0][2], toXYZD50.vals[1][2], toXYZD50.vals[2][2]));
-  }
+  tags.emplace_back(kTAG_rXYZ,
+                    write_xyz_tag(toXYZD50.vals[0][0], toXYZD50.vals[1][0], toXYZD50.vals[2][0]));
+  tags.emplace_back(kTAG_gXYZ,
+                    write_xyz_tag(toXYZD50.vals[0][1], toXYZD50.vals[1][1], toXYZD50.vals[2][1]));
+  tags.emplace_back(kTAG_bXYZ,
+                    write_xyz_tag(toXYZD50.vals[0][2], toXYZD50.vals[1][2], toXYZD50.vals[2][2]));
 
   // Compute white point tag (must be D50)
   tags.emplace_back(kTAG_wtpt, write_xyz_tag(kD50_x, kD50_y, kD50_z));
 
   // Compute transfer curves.
-  if (tf != UHDR_CT_PQ) {
-    if (tf == UHDR_CT_HLG) {
-      std::vector<uint8_t> trc_table;
-      trc_table.resize(kTrcTableSize * 2);
-      for (uint32_t i = 0; i < kTrcTableSize; ++i) {
-        float x = i / (kTrcTableSize - 1.f);
-        float y = hlgOetf(x);
-        y *= compute_tone_map_gain(tf, y);
-        float_to_table16(y, &trc_table[2 * i]);
-      }
-
-      tags.emplace_back(kTAG_rTRC,
-                        write_trc_tag(kTrcTableSize, reinterpret_cast<uint8_t*>(trc_table.data())));
-      tags.emplace_back(kTAG_gTRC,
-                        write_trc_tag(kTrcTableSize, reinterpret_cast<uint8_t*>(trc_table.data())));
-      tags.emplace_back(kTAG_bTRC,
-                        write_trc_tag(kTrcTableSize, reinterpret_cast<uint8_t*>(trc_table.data())));
-    } else if (tf == UHDR_CT_SRGB) {
-      tags.emplace_back(kTAG_rTRC, write_trc_tag(kSRGB_TransFun));
-      tags.emplace_back(kTAG_gTRC, write_trc_tag(kSRGB_TransFun));
-      tags.emplace_back(kTAG_bTRC, write_trc_tag(kSRGB_TransFun));
-    } else if (tf == UHDR_CT_LINEAR) {
-      tags.emplace_back(kTAG_rTRC, write_trc_tag(kLinear_TransFun));
-      tags.emplace_back(kTAG_gTRC, write_trc_tag(kLinear_TransFun));
-      tags.emplace_back(kTAG_bTRC, write_trc_tag(kLinear_TransFun));
-    }
+  if (tf == UHDR_CT_SRGB) {
+    tags.emplace_back(kTAG_rTRC, write_trc_tag(kSRGB_TransFun));
+    // Use empty data to indicate that the entry should use the previous tag's
+    // data.
+    tags.emplace_back(kTAG_gTRC, make_empty());
+    // Use empty data to indicate that the entry should use the previous tag's
+    // data.
+    tags.emplace_back(kTAG_bTRC, make_empty());
   }
 
+  // Chroma adaptation matrix
+  tags.emplace_back(kTAG_chad, write_chad_tag());
```

Comment:
1. Does it need to happen for all cases or just display P3
2. Even for display P3 I think our data selection of kDisplayP3 has already taken the white point shifting into account, so this may not be needed.